### PR TITLE
17 add request for actors to show actors objects not only ids

### PR DESCRIPTION
**Pull request by AndreevSV**
Task done

### CODE REVIEW

File: movies/service.js [hunk 1]
```diff
@@ -80,6 +92,38 @@ export const deleteMovie = async (id) => {
...
+    movies.map(async (movie) => {
+      let ids = movie.actors;
+      const actorsArr = await constructActorsArr(ids);
+      movie.actors = actorsArr;
+      return movie;
+    })
+  );
+  return newMovies;
+};
+
```
Comment (MaleryValery) [inline movies/service.js:126]: все равно вложенность массива в массиве, но без лишних запросов
const constructNewMoviesArr = async (movies) => {
  if (!movies.length) return [];

  const actorIds = movies.map((movie) => movie.actors).flat();
  const actorsBD = await ActorModel.find({ _id: { $in: actorIds } });
  const actorsMap = new Map(actorsBD.map((actorsBD) => [actorsBD._id.toString(), actorsBD]));

  movies.forEach((movie) => {
    movie.actors = movie.actors.map((actor) => actorsMap.get(actor));
  });
  return movies;
};

File: movies/service.js [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 export const fetchMovieById = async (id) => {
   try {
-    return await MovieModel.findById(id);
+    const movie = await MovieModel.findById(id);
+    if (!movie) {
+      logger.error('Movie not found');
+      return null;
+    }
+    let ids = movie.actors;
+    const actorsArr = await constructActorsArr(ids);
```
Comment (MaleryValery) [inline movies/service.js:26]: const actorsArr = await ActorModel.find({ _id: { $in: ids } });

**Review by MaleryValery — CHANGES_REQUESTED** [review]

File: movies/service.js (hunk 2 above)
Comment (AndreevSV) [reply movies/service.js:26]: It's not allowed to make a direct request to DB from one model to another model. In this case use only methods of service from another object. Services talk just between each other and service can request only model that it was made for

File: movies/service.js (hunk 1 above)
Comment (AndreevSV) [reply movies/service.js:126]: Same answer like below. You are not allowed  to use direct request from one model to another model in service. You should use just another service.

**Review by MaleryValery — APPROVED** [review]

**Pull request merged**